tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/Net/ASM/GraphQLSecurity.cs

```
@@ -0,0 +1,176 @@
+                    }
+
+                    value = GetValue(toDuckValue);
+
+                    object GetValue(object duckVal)
+                    {
+                        // if Value is VariableReference
+                        if (duckVal.TryDuckCast<GraphQLObjectValueProxy>(out var objFieldValue))
+                        {
+                            var dic = new Dictionary<string, object>();
+                            // return objFieldValue.ObjectFields.Select(GetValue).ToList();
+
+                            foreach (var obj in objFieldValue.ObjectFields)
+                            {
+                                if (obj.TryDuckCast<GraphQLObjectFieldProxy>(out var objField))
+                                {
+                                    dic.Add(objField.NameNode.Name, GetValue(objField.Value));
+                                }
+                            }
+
+                            return dic;
+                        }
+                        else if (duckVal.TryDuckCast<GraphQLValuesProxy>(out var argValues))
+                        {
+                            return argValues.Values.Select(GetValue).ToList();
+                        }
+                        else if (duckVal.TryDuckCast<GraphQLValueProxy>(out var argValue))
+                        {
+                            return GetValue(argValue.Value);
+                        }
+                        else if (duckVal.TryDuckCast<VariableReferenceProxy>(out var variableRef))
+                        {
+                            return GetVariableValue(context, variableRef.Name);
+                        }
+
+                        return duckVal.ToString();
+                    }
+                }
+                else
+                {
+                    // An unknown type of Argument is in the AST
+                    break;
+                }
+
+                resolverArguments.Add(name, value);
+            }
+            catch
```

Empty catch block again.